NowPlaying.Core/Api/RequestsManager.cs

```diff
@@ -1,46 +1,42 @@
 using System.Threading;
-using System.Net;
-using System.Text;
+using System.Threading.Tasks;
+using System.Net.Http;
+using System.Collections.Generic;
 using Newtonsoft.Json;
 
 namespace NowPlaying.Core.Api
 {
     public abstract class RequestsManager
     {
-        private WebClient CreateWebClient()
-        {
-            return new WebClient() { Encoding = Encoding.UTF8 };
-        }
+        protected readonly HttpClient client = new HttpClient { Timeout = Timeout.InfiniteTimeSpan };
```

Comment:
нужно сделать так, чтобы `RequestsManager` принимал `IHttpClient` через конструктор

а `HttpClient` пусть создаётся где-нибудь в другом месте

Reply:
Типа представь если у нас не только `RequestsManager` мог бы отправлять хттп-реквесты, но и кто-нибудь другой.
По-хорошему все они должны юзать один и тот же инстанс `HttpClient`.

```
HttpClient is intended to be instantiated once and re-used throughout the life of an application.
Instantiating an HttpClient class for every request will exhaust the number of sockets available under heavy loads.
This will result in SocketException errors.
```


https://docs.microsoft.com/en-gb/dotnet/api/system.net.http.httpclient?view=netframework-4.7.1#examples
![image](https://user-images.githubusercontent.com/17011936/81114573-f92b8a00-8f2a-11ea-9eed-c34a5ac94a12.png)